lib/process-services-cloud/src/lib/services/variable-mapper.sevice.ts

```diff
@@ -0,0 +1,91 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Injectable } from '@angular/core';
+import { DataColumn, DataColumnType } from '@alfresco/adf-core';
+import { ProcessListDataColumnCustomData } from '../models/data-column-custom-data';
+import { ProcessInstanceVariable, WithVariablesMap } from '../models/process-instance-variable.model';
+
+@Injectable({ providedIn: 'root' })
+export class VariableMapperService {
+    mapVariablesByColumnTitle <T extends { variables?: ProcessInstanceVariable[] }>(
+        instancesList: T[] = [],
+        columnsSchema: DataColumn<ProcessListDataColumnCustomData>[] = []
+    ): Array<WithVariablesMap<T>> {
+        const columnsByVariableId = this.mapColumnKeysByVariableId(columnsSchema);
+
+        const rowsViewModel = instancesList.map((instance) => {
```

Comment:
can we add type here?

Reply:
added `instancesList.map<WithVariablesMap<T>>((instance) => {`